File: backend/src/modules/instrutor/entity/Instrutor.js
class Instrutor {
  constructor({ id_instrutor, nome, telefone, email }) {
    this.id_instrutor = id_instrutor;
    this.nome = nome;
    this.telefone = telefone;
    this.email = email;
  }

  validate() {
    const errors = [];
    if (!this.nome) errors.push('Nome é obrigatório');
    if (!this.email) errors.push('Email é obrigatório');
    return errors;
  }
}

module.exports = Instrutor;

File: backend/src/main/java/com/vitalativa/entity/Instrutor.java
package com.vitalativa.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

@Entity
@Table(name = "instrutor")
public class Instrutor {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_instrutor")
    private Integer idInstrutor;
    
    @NotBlank(message = "Nome é obrigatório")
    private String nome;
    
    private String telefone;
    
    @NotBlank(message = "Email é obrigatório")
    @Email(message = "Email inválido")
    private String email;
    
    public Instrutor() {}
    
    public Integer getIdInstrutor() { return idInstrutor; }
    public void setIdInstrutor(Integer idInstrutor) { this.idInstrutor = idInstrutor; }
    public String getNome() { return nome; }
    public void setNome(String nome) { this.nome = nome; }
    public String getTelefone() { return telefone; }
    public void setTelefone(String telefone) { this.telefone = telefone; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
}